ivadomed/loader/utils.py

```diff
@@ -835,3 +844,36 @@ def write_derivatives_dataset_description(self, path_data):
             f = open(deriv_desc_file, 'w')
             f.write('{"Name": "Example dataset", "BIDSVersion": "1.0.2", "PipelineDescription": {"Name": "Example pipeline"}}')
             f.close()
+
+
+def get_file_extension(filename):
+    """ Get file extension if it is supported
+    Args:
+        filename (str): Path of the file.
+
+    Returns:
+        str: File extension
+    """
+    # Find the first match from the list of supported file extensions
+    extension = next((ext for ext in EXT_LST if filename.lower().endswith(ext)), None)
+
+    # TODO: remove "ome" from condition when implementing OMETIFF support (#739)
+    if (not extension) or ("ome" in extension):
+        raise RuntimeError("The input file type of '{}' is not supported".format(filename))
```

Comment:
This is a good question with a quite long answer :)

1. I don't use `pathlib.Path.suffixes` because we have to account for files with two part extensions (ex: `nii.gz`), for files with similar extensions (we don't want to mistake a `tiff` with an `ome.tiff`) and for files that could include another dot in the filenames (could happen when calling directly `segment_volume`, ex: with `sct_deepseg`). This is why we use an ordered list of supported extensions and a `str.endswith`.

2. I agree that the validation should not be done everytime and I suggest to move the validation (`if (not extension) or ("ome" in extension):`) in `SegmentationPair --> read_file`. This way, the validation would be done only once when the file is first loaded.

3. I cannot do this in the `BidsDataframe` because some entry points don't use it. Basically, here are the different entry points:

- command train/test:  `BidsDataframe --> BidsDataset --> MRI2DSegmentationDataset --> SegmentationPair`
- command segment: `BidsDataframe  --> segment_volume --> MRI2DSegmentationDataset --> SegmentationPair`
- sct_deepseg (and soon ADS): --> `segment_volume --> MRI2DSegmentationDataset --> SegmentationPair`

Same 3 options with 3D loader and HDF5 (adaptative) loader.
The one thing that they have in common is that they all go through `SegmentationPair --> read_file`. To avoid any code duplication, I think this is the best place to do the check.

Let me know what you think!

Reply:
> * sct_deepseg (and soon ADS): --> `segment_volume --> MRI2DSegmentationDataset --> SegmentationPair`

Aha, this is the part that I was missing. I had been assuming that everything went through `BidsDataframe`. Thank you giving such an in-depth answer! :tada:

Reply:
I moved the validation from `get_file_extension` to `read_file` after fetching the extension.